### ISSUE TITLE
The return value of the `sort `function is not used.

### ISSUE THREAD

**Issue by prim** [microsoft/pyright#7474]
If you have a question about a behavior that you’re seeing in pyright, consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Is your feature request related to a problem? Please describe.**
A clear and concise description of the problem.

```python
def f():
  l = []
  sorted(l)
```

I'm hoping pyright will tell me that the return value of the `sorted` function is not used.

**Describe the solution you’d like**
A clear and concise description of the solution you would like to see.

**Comment by erictraut** [comment]
There are many legitimate reasons in Python why you would want to ignore the return value of a function, so I don't see how such a rule could be implemented without generating a significant number of false positives.

Also, such a check is arguably not the job of a type checker. It's more of a linter sort of rule.

For these reasons, I'm going to reject this enhancement request and close it.
